src/sync/impl/sync_metadata.cpp

```diff
@@ -426,6 +449,79 @@ std::shared_ptr<Realm> SyncMetadataManager::get_realm() const
     return realm;
 }
 
+void SyncMetadataManager::set_app_metadata(const std::string& deployment_model,
+                                           const std::string& location,
+                                           const std::string& hostname,
+                                           const std::string& ws_hostname) const
+{
+    auto realm = get_realm();
+    auto& schema = m_app_metadata_schema;
+    
+    realm->begin_transaction();
+    
+    auto table = ObjectStore::table_for_object_type(realm->read_group(), c_sync_app_metadata);
+    auto obj = table->create_object();
```

Comment:
In theory there should only ever been one app metadata object. It is initialized once on first app usage, then never updated.

Reply:
I guess my concern was that I saw a comment from Eric that it might update, then we might eventually add that logic and forget to update this. Which will introduce _really_ subtle bugs.

Reply:
Adding a primary key field and assign it a single value, should make this scenario impossible.

Reply:
What would the primary key be?

Reply:
Any value would work, but `1` ? That way you can use `table->find_first_int(ColKey(id_col), 1)`